pkg/reconciler/containersource/containersource_test.go

```diff
@@ -228,6 +233,79 @@ func TestAllCases(t *testing.T) {
 					), &conditionTrue)),
 				),
 			}},
+		}, {
+			Name: "OIDC: creates OIDC service account",
+			Key:  testNS + "/" + sourceName,
+			Ctx: feature.ToContext(context.Background(), feature.Flags{
+				feature.OIDCAuthentication: feature.Enabled,
+			}),
+			Objects: []runtime.Object{
+				NewContainerSource(sourceName, testNS,
+					WithContainerSourceUID(sourceUID),
+					WithContainerSourceSpec(makeContainerSourceSpec(sinkDest)),
+					WithContainerSourceObjectMetaGeneration(generation),
+				),
+				makeSinkBinding(NewContainerSource(sourceName, testNS,
+					WithContainerSourceSpec(makeContainerSourceSpec(sinkDest)),
+					WithContainerSourceUID(sourceUID),
+				), &conditionTrue),
+				makeDeployment(NewContainerSource(sourceName, testNS,
+					WithContainerSourceSpec(makeContainerSourceSpec(sinkDest)),
+					WithContainerSourceUID(sourceUID),
+				), &conditionTrue),
+			},
+			WantErr: false,
+			WantStatusUpdates: []clientgotesting.UpdateActionImpl{{
+				Object: NewContainerSource(sourceName, testNS,
+					WithContainerSourceUID(sourceUID),
+					WithContainerSourceSpec(makeContainerSourceSpec(sinkDest)),
+					WithContainerSourceObjectMetaGeneration(generation),
+					WithInitContainerSourceConditions,
+					WithContainerSourceStatusObservedGeneration(generation),
+					WithContainerSourceOIDCIdentityCreatedSucceeded(),
+					WithContainerSourcePropagateSinkbindingStatus(makeSinkBindingStatus(&conditionTrue)),
+					WithContainerSourcePropagateReceiveAdapterStatus(makeDeployment(NewContainerSource(sourceName, testNS,
+						WithContainerSourceSpec(makeContainerSourceSpec(sinkDest)),
+						WithContainerSourceUID(sourceUID),
+					), &conditionTrue)),
```

Comment:
```suggestion
				Object: NewContainerSource(sourceName, testNS,
					WithContainerSourceUID(sourceUID),
					WithContainerSourceSpec(makeContainerSourceSpec(sinkDest)),
					WithContainerSourceObjectMetaGeneration(generation),
					WithInitContainerSourceConditions,
					WithContainerSourceStatusObservedGeneration(generation),
					WithContainerSourcePropagateSinkbindingStatus(makeSinkBindingStatus(&conditionTrue)),
					WithContainerSourcePropagateReceiveAdapterStatus(makeDeployment(NewContainerSource(sourceName, testNS,
						WithContainerSourceSpec(makeContainerSourceSpec(sinkDest)),
						WithContainerSourceUID(sourceUID),
					), &conditionTrue)),
					WithContainerSourceOIDCIdentityCreatedSucceeded(),
					WithContainerSourceOIDCServiceAccountName(makeContainerSourceOIDCServiceAccount().Name),
				),
```
seems you need to call the `WithContainerSourceOIDC*` functions latest, otherwise it gets overridden. Also you need to call `WithContainerSourceOIDCServiceAccountName()`.

When the ContainerSource reconciles successfully, it publishes a kubernetes event:

```
WantEvents: []string{
	Eventf(corev1.EventTypeNormal, sourceReconciled, `ContainerSource reconciled: "%s/%s"`, testNS, sourceName),
},
```